client_test.go

```diff
@@ -0,0 +1,2068 @@
+			`, tc.statement))
+
+				return err
+			})
+
+			assert.Error(t, gotScriptErr)
+		})
+	}
+}
+
+func TestClientIsSupportedType(t *testing.T) {
+	t.Parallel()
+
+	t.Run("table tests", func(t *testing.T) {
+		t.Parallel()
+
+		tests := []struct {
+			name    string
+			offset  int
+			args    []interface{}
+			wantErr bool
+		}{
+			{
+				name:    "string is a supported type",
+				offset:  1,
+				args:    []interface{}{"foo"},
+				wantErr: false,
+			},
+			{
+				name:    "int is a supported type",
+				offset:  1,
+				args:    []interface{}{int(123)},
+				wantErr: false,
+			},
+			{
+				name:    "int64 is a supported type",
+				offset:  1,
+				args:    []interface{}{int64(123)},
+				wantErr: false,
+			},
+			{
+				name:    "float64 is a supported type",
+				offset:  1,
+				args:    []interface{}{float64(123)},
+				wantErr: false,
+			},
+			{
+				name:    "bool is a supported type",
+				offset:  1,
+				args:    []interface{}{bool(true)},
+				wantErr: false,
+			},
+			{
+				name:    "multiple identical types args are supported",
+				offset:  1,
+				args:    []interface{}{int(123), int(456)},
+				wantErr: false,
+			},
+			{
+				name:    "multiple mixed types args are supported",
+				offset:  1,
+				args:    []interface{}{int(123), "foo", bool(true)},
+				wantErr: false,
+			},
+			{
+				name:    "slice[T] is not a supported type",
+				offset:  1,
+				args:    []interface{}{[]string{"1", "2", "3"}},
+				wantErr: true,
+			},
+			{
+				name:    "multiple mixed valid and invalid types args are not supported",
+				offset:  1,
+				args:    []interface{}{int(123), []string{"1", "2", "3"}},
+				wantErr: true,
+			},
+		}
+		for _, tt := range tests {
+			tt := tt
+
+			t.Run(tt.name, func(t *testing.T) {
+				t.Parallel()
+
+				c := &Client{} // nolint:exhaustruct
+				gotErr := c.isSupportedType(tt.offset, tt.args...)
+				assert.Equal(t,
+					tt.wantErr,
+					gotErr != nil,
+					"Client.isSupportedType() error = %v, wantErr %v", gotErr, tt.wantErr,
+				)
+			})
+		}
+	})
+
+	t.Run("offset is respected in the error message", func(t *testing.T) {
+		t.Parallel()
+
+		c := &Client{} // nolint:exhaustruct
+
+		gotErr := c.isSupportedType(3, int(123), []string{"1", "2", "3"})
+
+		assert.Error(t, gotErr)
+		assert.Contains(t, gotErr.Error(), "argument at index: 4")
+	})
+}
+
+// testSetup is a helper struct holding components
+// necessary to test the redis client, in the context
+// of the execution of a k6 script.
+type testSetup struct {
+	rt      *goja.Runtime
+	state   *lib.State
+	samples chan metrics.SampleContainer
+	ev      *eventloop.EventLoop
+	// miniredis is a mock redis server, in the
+	// spirit of golang's httptest.
+	redis *miniredis.Miniredis
+}
+
+// newTestSetup initializes a new test setup.
+// It prepares a test setup with a mocked redis server and a goja runtime,
+// and event loop, ready to execute scripts as if being executed in the
+// main context of k6.
```

Comment:
Haven't looked through all the code but in general it might be better to just have a method on testSetup to move from init context to vu context or somethign like that :shrug: